### ISSUE TITLE
Holding delete key with delete word before cursor in the settings text field causes crash

### ISSUE THREAD

**Issue by whatevenisthat** [florisboard/florisboard#2001]
#### Short description
When you have delete key long press set to delete word before cursor, holding the delete key in the Florisboard Settings app while the text field is empty causes Florisboard the crash.  This crash is new to 0.4.0-alpha02 and also doesn't seem to occur in other apps I've tried.

#### Steps to reproduce
1. Set delete key long press to delete word before cursor.
2. Open the keyboard by tapping the 'Try out your setup' text field.
3. Hold the delete key.
4. See crash.

#### Environment information
- FlorisBoard [0.4.0-alpha02](https://github.com/florisboard/florisboard/releases/v0.4.0-alpha02) (88)
- Device: HMD Global Nokia 7.1 (Crystal_00WW)
- Android: 10 (cn=Q sdk=29)

#### Attached logs and stacktrace files
<details>
<summary>Detailed info (Debug log header)</summary>

```
======= SYSTEM INFO =======
Time                : 2022-08-03T21:51:15.426Z
Manufacturer        : HMD Global
Model               : Nokia 7.1
Product             : Crystal_00WW
Android             : 10 (cn=Q sdk=29) [00WW_4_15X]
ABIs                : [arm64-v8a, armeabi-v7a, armeabi]
Memory              : 1.91 GiB (71.47% used, 2.67 GiB max)
Font scale          : 1.0
Locales             : en-GB

======= APP INFO =======
Package             : dev.patrickgold.florisboard.beta
Name                : FlorisBoard Beta
Version             : 0.4.0-alpha02 (88)
Build type          : beta
Build commit hash   : 53ab0a3fa016d71206acd9dcc076dde913782640
Java heap memory    : 5.92 MiB (2.31% used, 256.00 MiB max)
Native heap memory  : 21.99 MiB (77.80% used, 28.27 MiB max)

======= FEATURE CONFIG =======
Smartbar enabled            : true
Suggestions enabled         : true
Inline autofill enabled     : true
Glide enabled               : true
Internal clipboard enabled  : true

```
</details>

<details>
<summary>1659563473227.stacktrace</summary>

```
java.lang.IllegalArgumentException: offset(-1) is out of bounds [0, 0]
	at androidx.compose.ui.text.MultiParagraph.requireIndexInRangeInclusiveEnd(MultiParagraph.kt:9)
	at androidx.compose.ui.text.TextLayoutResult.getWordBoundary--jx7JFs(TextLayoutResult.kt:2)
	at androidx.compose.foundation.text.selection.BaseTextPreparedSelection.getPrevWordOffset(TextPreparedSelection.kt:4)
	at androidx.compose.foundation.text.TextFieldKeyInput$process$2$5.invoke(TextFieldKeyInput.kt:5)
	at androidx.compose.foundation.text.selection.TextFieldPreparedSelection.deleteIfSelectedOr(TextPreparedSelection.kt:3)
	at androidx.compose.foundation.text.TextFieldKeyInput$process$2.invoke(TextFieldKeyInput.kt:95)
	at androidx.compose.foundation.text.TextFieldKeyInputKt$textFieldKeyInput$2$1.invoke(TextFieldKeyInput.kt:28)
	at androidx.compose.ui.input.key.KeyInputModifier.propagateKeyEvent-ZmokQxo(KeyInputModifier.kt:3)
	at androidx.compose.ui.platform.AndroidComposeView.dispatchKeyEvent(AndroidComposeView.android.kt:20)
	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1922)
	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1922)
	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1922)
	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1922)
	at com.android.internal.policy.DecorView.superDispatchKeyEvent(DecorView.java:453)
	at com.android.internal.policy.PhoneWindow.superDispatchKeyEvent(PhoneWindow.java:1839)
	at android.app.Activity.dispatchKeyEvent(Activity.java:3953)
	at androidx.core.app.ComponentActivity.superDispatchKeyEvent(Unknown Source:0)
	at androidx.core.app.ComponentActivity.dispatchKeyEvent(ComponentActivity.java:4)
	at com.android.internal.policy.DecorView.dispatchKeyEvent(DecorView.java:367)
	at android.view.ViewRootImpl$ViewPostImeInputStage.processKeyEvent(ViewRootImpl.java:5411)
	at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:5279)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4786)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4839)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4805)
	at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4945)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4813)
	at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:5002)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4786)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4839)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4805)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4813)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4786)
	at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:7509)
	at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:7478)
	at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:7439)
	at android.view.ViewRootImpl$ViewRootHandler.handleMessage(ViewRootImpl.java:4559)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7397)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:935)

```
</details>

**Comment by patrickgold** [comment]
Does this happen in any other text field than the "Try out your setup field" too?

**Comment by Keepkeepzz** [comment]
It seems to happen with all the text fields in the Florisboard settings apps. I haven't been able to get this crash to occur in other apps though.

**Comment by patrickgold** [comment]
Hmm, so I think this is connected with the UI library upgrade to Compose 1.2. I can fully reproduce this bug, will check tomorrow if this is a bug of mine or of Compose.

**Comment by patrickgold** [comment]
See what I found: https://issuetracker.google.com/issues/208655565

Issue is indeed caused by COmpose, and it also seems that they fixed this crash in 1.3.0-alpha02. so I might upgrade to that lib, will see later.
